client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js

```diff
@@ -127,8 +130,46 @@ export default function WPCheckout( {
 		setShouldShowContactDetailsValidationErrors,
 	] = useState( false );
 
+	const emailTakenLoginRedirectMessage = ( emailAddress ) => {
+		const redirectTo = '/checkout/no-site?cart=no-user';
+		const isNative = config.isEnabled( 'login/native-login-links' );
+		const loginUrl = login( { redirectTo, emailAddress, isNative } );
+		const loginRedirectMessage = translate(
+			'Choose a different email address. This one is not available. If this is you {{a}}log in now{{/a}}.',
+			{
+				components: {
+					a: <a href={ loginUrl } />,
+				},
+			}
+		);
+		return loginRedirectMessage;
+	};
+
 	const validateContactDetailsAndDisplayErrors = async () => {
 		debug( 'validating contact details with side effects' );
+		if ( isLoggedOutCart ) {
+			const email = contactInfo.email?.value;
```

Comment:
If the optional chaining falls through, then `email` will be `undefined` here, and that `undefined` will be passed through the rest of the functions above. That might actually be totally fine, but if it's not, we could set a fallback value here like:

```js
const email = contactInfo.email?.value ?? '';
```

I don't know if it's an issue so I'll leave it to you to decide.

Reply:
Ah yes, it doesn't lead to an error but good to have it this way. I've made this change.